code/game/atoms.dm

```diff
@@ -165,8 +165,27 @@
 			found += A.search_contents_for(path,filter_path)
 	return found
 
-
-
+/*
+ * Recursively searches all atom contents (including contents contents and so on.)
+ * 
+ * ARGS: paths (array) - only count atoms of one the provided paths.
+ * RETURNS: A number of all contents found.
+*/
+/atom/proc/get_all_contents_count(var/paths)
+	var/count = 0
+	for(var/atom/A in src.contents)
+		if(paths)
+			var/found = FALSE
+			for(var/path in paths)
+				if(istype(A, path))
+					found = TRUE
+					break
+			if(!found)
+				continue
+		count++
+		if(A.contents.len)
+			count += A.get_all_contents_count()
```

Comment:
BYOND doesn't do too well with recursion, should instead unroll into a loop

Reply:
That should be true too for line 165. I can update both, can I get an example of what that would look like? I'm not super familiar with how to flatten that data.

Reply:
Line 513 of code/_helpers/unsorted.dm uses recursion for GetAllContents()
Line 165 of code/game/atoms.dm uses recursion for path searching contents.

Just additional places where this critique applies, and maybe out of scope of this PR for this requested change to be enacted. Especially because I don't know how to "unroll" it without fancy features present in modern languages.

Reply:
````
	var/count = 0
	var/list/counting_atoms = contents.Copy() + src
	while(length(counting_atoms))
		var/atom/movable/thing = counting_atoms[1]
		counting_atoms -= thing
		counting_atoms |= thing.contents
		for(var/path in paths)
			if(istype(thing, path))
				count++
				break
````

Reply:
This only goes one level deep and would break my code.

Reply:
No it doesn't?